app/controllers/concerns/cookie_concern.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+module CookieConcern
+  extend ActiveSupport::Concern
+
+  def update_cookies
+    cookie_params = params.permit(:cookies_preference, :hide_banner)
+
+    render(partial: 'layouts/cookie_banner/hide') && return if params['hide_banner'] == 'true'
+
+    update_analytics_cookies(cookie_params[:cookies_preference])
+
+    render(partial: 'layouts/cookie_banner/success',
+           locals: { analytics_cookies_accepted: cookies[:analytics_cookies_set] })
+  end
+
+  def set_default_cookies
+    return set_default_analytics_cookies unless cookies[:cookies_preferences_set]
+
+    @analytics_cookies_accepted = ActiveModel::Type::Boolean.new.cast(cookies[:analytics_cookies_set])
+    @cookie_preferences_set = show_hide_cookie_banners
```

Comment:
this is always true as you have a guard up about. :) (line 18)

Reply:
First time you view the website or when the cookies expire this will be false

Reply:
no I mean the value will always be true if it gets to this line of code... so could replace this line with
```
@cookie_preferences_set = show_hide_cookie_banners
```